### PR TITLE
make sure OpenSSL is initialized before Tls13Supported code runs

### LINKED ISSUE #61891: CipherSuitesPolicy constructor throws PlatformNotSupportedException on Linux with OpenSSL 1.1.1
### Description

According to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.ciphersuitespolicy.-ctor?view=net-6.0#System_Net_Security_CipherSuitesPolicy__ctor_System_Collections_Generic_IEnumerable_System_Net_Security_TlsCipherSuite__), it should be possible to use the CipherSuitesPolicy class on Linux systems with OpenSSL 1.1.1 or higher or macOS. However, when using .NET 6 on Linux with OpenSSL 1.1.1, PlatformNotSupportedException is thrown. The same code works fine when using .NET 5

### Reproduction Steps

Run the following program:
```c#
using System.Net.Security;

namespace csptest
{
    public static class Program
    {
        public static void Main()
        {
            var cipherSuitesPolicy = new CipherSuitesPolicy(new TlsCipherSuite[] { });
        }
    }
}
```

### Expected behavior

Program completes successfully

### Actual behavior

Exception is thrown:

```
Unhandled exception. System.PlatformNotSupportedException: CipherSuitesPolicy is not supported on this platform.
   at System.Net.Security.CipherSuitesPolicyPal..ctor(IEnumerable`1 allowedCipherSuites)
   at System.Net.Security.CipherSuitesPolicy..ctor(IEnumerable`1 allowedCipherSuites)
   at csptest.Program.Main() in /home/gth/csptest/Program.cs:line 9
```

### Regression?

Seems like it, as it works fine on .NET 5

### Known Workarounds

_No response_

### Configuration

**.NET**
.NET 6
```
dotnet --list-sdks
6.0.100 [/usr/share/dotnet/sdk]
dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

***OpenSSL***
OpenSSL 1.1.1f
```
openssl version
OpenSSL 1.1.1f  31 Mar 2020
```

***OS***
Ubuntu 20.04.3 LTS x86_64 (completely clean install)
```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```
Same issue also occurs on Ubuntu 21.10

### Other information

_No response_

### PR DESCRIPTION
**Pull request by wfurt**
replacement for  #62037

This is regression caused by #46640. Tls13Supported can be called before static constructor runs. It was OK in 5.0 since the OpenSSL library was initialized via

```c
__attribute__((constructor))
static void InitializeOpenSSLShim()
```
With 6.0, it will be initialized when Ssl us 
used but some properties like CiphersSuitePolicy can be used before in order to prepare everything needed.

fixes #61891.

### CODE REVIEW

File: src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs [hunk 1]
```diff
@@ -149,6 +149,16 @@ internal static partial class Ssl
         [GeneratedDllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslSetData")]
         internal static partial int SslSetData(IntPtr ssl, IntPtr data);
 
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_Tls13Supported")]
```
Comment (jkotas) [inline src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs:152]: This should stay as `GeneratedDllImport`

Comment (wfurt) [reply src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs]: fixed. It was cut & paste but it seems like my tree was not fully in sync.

File: src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs [hunk 2]
```diff
@@ -149,6 +149,15 @@ internal static partial class Ssl
         [GeneratedDllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslSetData")]
         internal static partial int SslSetData(IntPtr ssl, IntPtr data);
 
+        [GeneratedDllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_Tls13Supported")]
+        private static partial int Tls13SupportedImpl();
+
+        internal static class Capabilities
+        {
+            // needs separate type to be sure OpenSSL is initialized.
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs:157]:
```suggestion
            // needs separate type (separate static cctor) to be sure OpenSSL is initialized.
```

**Review by stephentoub — APPROVED** [review]

**Pull request merged**